projects/ziti-console-lib/src/lib/features/dynamic-widgets/port-ranges/port-ranges.component.ts

```diff
@@ -0,0 +1,94 @@
+import {Component, EventEmitter, Input, Output} from '@angular/core';
+import {isNumber} from 'lodash';
+
+@Component({
+  selector: 'lib-port-ranges',
+  templateUrl: './port-ranges.component.html',
+  styleUrls: ['./port-ranges.component.scss']
+})
+export class PortRangesComponent {
+
+  @Input() fieldValue: any = '';
+  @Output() fieldValueChanged: EventEmitter<any> = new EventEmitter<any>();
+  errors: any = {};
+  invalid: boolean = false;
+
+  onKeyDown(event?) {
+  }
+
+  get ranges(): any {
+    return {};
+  }
+
+  set ranges(ranges: any) {
+    const portRanges = ranges;
+  }
+
+  getProperties() {
+    if (!this.fieldValue) {
+      return [];
+    }
+    const ranges = [];
+    this.fieldValue.forEach((val: string) => {
+      const vals = val.split('-');
+      if (vals.length === 1) {
```

Comment:
can i assume that somewhere else, something has validated that 'val' can only be numbers and that those numbers are only from 1 to 65535 ? even still, i'd say this is the kind of code that should be separated on its own and reused as needed.